### ISSUE TITLE
Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Element not found.

### ISSUE THREAD

**Issue by raedwa01** [docker/for-win#815]
### Expected behavior
Docker will load
### Actual behavior
Docker crashes on startup
### Information
Diagnostic ID: 1FA415EA-EEEC-49E0-BE0B-51D78BC7261E/2017-06-12_11-06-39
### Steps to reproduce the behavior

  1. Start PC
  2. Docker Launches
  3. Crash

Here's my log file when starting:

Version: 17.03.1-ce-win12 (12058)
Channel: stable
Sha1: d1db12684b33b778cfc3012e92959e50eb486fcd
Started on: 2017/06/12 11:01:12.269
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 1703
Build: 15063
BuildLabName: 15063.0.amd64fre.rs2_release.170317-1834
File: C:\Users\Rob Edwards\AppData\Local\Docker\log.txt
CommandLine: "C:\Program Files\Docker\Docker\Docker for Windows.exe" 
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[11:01:12.358][GUI            ][Info   ] Starting...
[11:01:12.628][Tracking       ][Info   ] Crash report and usage statistics are enabled
[11:01:12.631][SegmentApi     ][Info   ] Usage statistic: appLaunched
[11:01:20.904][NamedPipeClient][Info   ] Sending Version()...
[11:01:20.914][NamedPipeClient][Info   ] Received response for Version
[11:01:20.915][SegmentApi     ][Info   ] Usage statistic: heartbeat
[11:01:19.024][BackendServer  ][Info   ] Started
[11:01:20.912][NamedPipeServer][Info   ] Version()
[11:01:20.913][NamedPipeServer][Info   ] Version done in 00:00:00.
[11:01:21.250][Updater        ][Info   ] Checking for updates on channel stable...
[11:01:21.252][NamedPipeClient][Info   ] Sending Start(Docker.Core.Settings)...
[11:01:21.253][NamedPipeServer][Info   ] Start(Docker.Core.Settings)
[11:01:21.256][PowerMode      ][Info   ] Stop
[11:01:21.257][HyperV         ][Info   ] Stop
[11:01:21.263][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[11:01:21.264][PowerShell     ][Info   ] Creating a Runspace Pool...
[11:01:21.345][Updater        ][Info   ] Local build 12058 is as good as the remote 12058 on channel stable
[11:01:21.394][PowerShell     ][Info   ] Runspace Pool created (Min=1, Max=2)
[11:01:21.456][HyperV         ][Info   ] Script started at 11:01:21.452
[11:01:23.016][HyperV         ][Info   ] Module loaded at 11:01:23.016
[11:01:23.226][HyperV         ][Info   ] VM MobyLinuxVM does not exist
[11:01:23.226][HyperV         ][Debug  ] [stop] took 00:00:01.9679946 to run
[11:01:23.226][OptimizeDisk   ][Info   ] Optimize
[11:01:23.227][PowerShell     ][Info   ] Run script...
[11:01:23.241][Moby           ][Info   ] Stop
[11:01:23.252][HyperVGuids    ][Info   ] Installing GUIDs...
[11:01:23.252][PowerMode      ][Info   ] Start
[11:01:23.252][HyperV         ][Info   ] Create
[11:01:23.253][PowerShell     ][Info   ] Run script with parameters: -Create True -VhdPathOverride  -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\mobylinux.iso...
[11:01:23.253][HyperVGuids    ][Info   ] GUIDs installed
[11:01:23.254][Firewall       ][Info   ] Removing all existing rules...
[11:01:23.260][HyperV         ][Info   ] Script started at 11:01:23.260
[11:01:23.269][HyperV         ][Info   ] Module loaded at 11:01:23.269
[11:01:23.278][Firewall       ][Info   ] All existing rules are removed.
[11:01:23.278][Firewall       ][Info   ] Opening ports for C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe...
[11:01:23.286][Firewall       ][Info   ] Opening ports for SMB...
[11:01:23.311][Firewall       ][Info   ] Ports are opened
[11:01:23.418][HyperV         ][Info   ] Creating Switch: DockerNAT...
[11:01:24.576][HyperV         ][Info   ] Switch created.
[11:01:24.886][Linux          ][Error  ] Failed to start: Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Element not found.

. Let's retry
[11:01:24.887][PowerShell     ][Info   ] Run script...
[11:01:26.320][HyperV         ][Info   ] Hyper-V is running
[11:01:26.320][PowerMode      ][Info   ] Stop
[11:01:26.320][HyperV         ][Info   ] Stop
[11:01:26.320][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[11:01:26.331][HyperV         ][Info   ] Script started at 11:01:26.331
[11:01:26.338][HyperV         ][Info   ] Module loaded at 11:01:26.338
[11:01:26.345][HyperV         ][Info   ] VM MobyLinuxVM does not exist
[11:01:26.345][HyperV         ][Debug  ] [stop] took 00:00:00.0259872 to run
[11:01:26.345][OptimizeDisk   ][Info   ] Optimize
[11:01:26.345][PowerShell     ][Info   ] Run script...
[11:01:26.357][Moby           ][Info   ] Stop
[11:01:26.375][HyperV         ][Info   ] Destroy
[11:01:26.375][PowerShell     ][Info   ] Run script with parameters: -Destroy True -KeepVolume True...
[11:01:26.379][HyperV         ][Info   ] Script started at 11:01:26.379
[11:01:26.387][HyperV         ][Info   ] Module loaded at 11:01:26.387
[11:01:26.393][HyperV         ][Info   ] VM MobyLinuxVM does not exist
[11:01:26.394][HyperV         ][Info   ] Destroying Switch DockerNAT...
[11:01:26.593][HyperV         ][Info   ] Removing VM MobyLinuxVM...
[11:01:26.600][HyperV         ][Debug  ] [destroy] took 00:00:00.2259914 to run
[11:01:26.600][Firewall       ][Info   ] Closing ports...
[11:01:26.600][Firewall       ][Info   ] Removing all existing rules...
[11:01:26.652][Firewall       ][Info   ] All existing rules are removed.
[11:01:26.652][Firewall       ][Info   ] Ports are closed
[11:01:26.652][HyperVGuids    ][Info   ] Removing GUIDs...
[11:01:26.654][HyperVGuids    ][Info   ] GUIDs removed
[11:01:26.654][HyperV         ][Info   ] Create
[11:01:26.654][PowerShell     ][Info   ] Run script with parameters: -Create True -VhdPathOverride  -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\mobylinux.iso...
[11:01:26.670][HyperV         ][Info   ] Script started at 11:01:26.670
[11:01:26.677][HyperV         ][Info   ] Module loaded at 11:01:26.677
[11:01:26.723][HyperV         ][Info   ] Creating Switch: DockerNAT...
[11:01:27.199][HyperV         ][Info   ] Switch created.
[11:01:27.237][PowerMode      ][Info   ] Stop
[11:01:27.237][HyperV         ][Info   ] Stop
[11:01:27.237][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[11:01:27.243][HyperV         ][Info   ] Script started at 11:01:27.243
[11:01:27.252][HyperV         ][Info   ] Module loaded at 11:01:27.252
[11:01:27.258][HyperV         ][Info   ] VM MobyLinuxVM does not exist
[11:01:27.258][HyperV         ][Debug  ] [stop] took 00:00:00.0209991 to run
[11:01:27.258][OptimizeDisk   ][Info   ] Optimize
[11:01:27.258][PowerShell     ][Info   ] Run script...
[11:01:27.266][Moby           ][Info   ] Stop
[11:01:27.273][HyperV         ][Info   ] Destroy
[11:01:27.273][PowerShell     ][Info   ] Run script with parameters: -Destroy True -KeepVolume True...
[11:01:27.278][HyperV         ][Info   ] Script started at 11:01:27.278
[11:01:27.286][HyperV         ][Info   ] Module loaded at 11:01:27.286
[11:01:27.293][HyperV         ][Info   ] VM MobyLinuxVM does not exist
[11:01:27.293][HyperV         ][Info   ] Destroying Switch DockerNAT...
[11:01:27.481][HyperV         ][Info   ] Removing VM MobyLinuxVM...
[11:01:27.486][HyperV         ][Debug  ] [destroy] took 00:00:00.2129985 to run
[11:01:27.486][Firewall       ][Info   ] Closing ports...
[11:01:27.486][Firewall       ][Info   ] Removing all existing rules...
[11:01:27.503][Firewall       ][Info   ] All existing rules are removed.
[11:01:27.503][Firewall       ][Info   ] Ports are closed
[11:01:27.503][HyperVGuids    ][Info   ] Removing GUIDs...
[11:01:27.504][HyperVGuids    ][Info   ] GUIDs removed
[11:01:27.504][NamedPipeServer][Error  ] Unable to execute Start: Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Element not found.

    at Docker.Backend.HyperV.RunScript(String action, Dictionary`2 parameters)
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
[11:01:27.544][NamedPipeClient][Error  ] Unable to send Start: Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Element not found.

[11:01:27.545][Notifications  ][Error  ] Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Element not found.

[11:06:10.502][SegmentApi     ][Info   ] Usage statistic: actionMenuSettings
[11:06:10.652][NamedPipeClient][Info   ] Sending GetDefaultVhdxPath()...
[11:06:10.653][NamedPipeServer][Info   ] GetDefaultVhdxPath()
[11:06:10.690][NamedPipeClient][Info   ] Received response for GetDefaultVhdxPath
[11:06:10.689][NamedPipeServer][Info   ] GetDefaultVhdxPath done in 00:00:00.0359994.
[11:06:26.860][FeedbackSettings][Info   ] Open logs
[11:06:32.348][FeedbackSettings][Info   ] Open logs

**Comment by jasonbivins** [comment]
Hi @raedwa01   It looks like either the MobyLinuxVm or the DockerNat switch aren't getting created successfully.   Can you run this debug script in PS and let me know if you have any errors?
  Invoke-WebRequest https://aka.ms/Debug-ContainerHost.ps1 -UseBasicParsing | Invoke-Expression

Source and readme with failure descriptions are at https://github.com/Microsoft/Virtualization-Documentation/tree/live/windows-server-container-tools/Debug-ContainerHost

**Comment by raedwa01** [comment]
PS C:\WINDOWS\system32> Invoke-WebRequest https://aka.ms/Debug-ContainerHost.ps1 -UseBasicParsing | Invoke-Expression
Checking for common problems
Describing Windows Version and Prerequisites
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 528ms
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 106ms
 [+] Is not a build with blocking issues 22ms
 [-] Has 'Containers' feature installed 42.96s
   Expected: {Enabled}
   But was:  {Disabled}
   23:             (Get-WindowsOptionalFeature -Online -FeatureName Containers).State | Should Be "Enabled"
   at <ScriptBlock>, <No file>: line 23
Describing Docker is installed
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  58ms
 [+] Service is running 19ms
 [+] Docker.exe is in path 2.11s
 [+] Docker is registered in the EventLog service 53ms
Describing User has permissions to use Docker daemon
 [+] docker.exe should not return access denied 34ms
Describing Windows container settings are correct
 [+] Do not have DisableVSmbOplock set to 1 27ms
 [+] Do not have zz values set 22ms
 [+] Do not have FDVDenyWriteAccess set to 1 17ms
Describing The right container base images are installed
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.27/images/json: open //./pipe/docker_engine: The syst
em cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevate
d to connect. This error may also indicate that the docker daemon is not running.
 [-] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should be installed 67ms
   ValidationMetadataException: The argument is null or empty. Provide an argument that is not null or empty, and then t
ry the command again.
   ParameterBindingValidationException: Cannot validate argument on parameter 'Property'. The argument is null or empty.
 Provide an argument that is not null or empty, and then try the command again.
   at <ScriptBlock>, <No file>: line 115
Describing Container network is created
 [+] At least one local container network is available 1.04s
 [-] At least one NAT, Transparent, or L2Bridge Network exists 33ms
   Expected {0} to be greater than {0}
   206:       $totalnets | Should BeGreaterThan 0
   at <ScriptBlock>, <No file>: line 206
 [-] NAT Network's vSwitch is internal 28ms
   Expected: {Internal}
   But was:  {}
   211:       $switchType | Should Be "Internal"
   at <ScriptBlock>, <No file>: line 211
 [+] A Windows NAT is configured if a Docker NAT network exists 247ms
 [-] Specified Network Gateway IP for NAT network is assigned to Host vNIC 26ms
   Expected: value to not be empty
   225:       $natGatewayIP | Should Not BeNullOrEmpty
   at <ScriptBlock>, <No file>: line 225
 [-] NAT Network's internal prefix does not overlap with external IP' 25ms
   Expected {0} to be greater than {0}
   252:         $hostips.Count | Should BeGreaterThan 0
   at <ScriptBlock>, <No file>: line 252
Showing output from: docker info

Showing output from: docker version
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64

Showing output from: docker network ls

Getting Warnings & errors in the Windows event logs from the last 24 hours
Logs saved to C:\WINDOWS\system32\logs_20170612-135818.csv


Getting Docker for Windows daemon logs from the last execution
    Note: More logs are available at C:\Users\Rob Edwards\AppData\Local\Docker. Only showing the latest.
PS C:\WINDOWS\system32>
PS C:\WINDOWS\system32>

**Comment by raedwa01** [comment]
I ran the change to enable containers manually.  So that error is gone from my next run, but the rest still remain.

**Comment by jasonbivins** [comment]
It does looke like there was a problem with getting the DockerNat internal switch created. 
Can you check to see if there is an existing DockerNat virtual switch?
Have you tried to re-install D4W  ?  Try re-running the installer as an administrator now that you have containers enabled.

**Comment by raedwa01** [comment]
I uninstalled, rebooted, launched the command prompt as admin and ran InstallDocker.msi.  Still doing the same thing.

**Comment by jasonbivins** [comment]
Can you try to create an internal switch in Hyper-v ?  Do you have an existing vEthernet connection called DockerNat  ?

![image](https://user-images.githubusercontent.com/28021664/27095322-3b1d588a-503b-11e7-90c6-d71bd128d30b.png)

**Comment by raedwa01** [comment]
I created an internal switch in HyperV (you can see it in my screen shot below.  The interesting thing is that I did not have a DockerNAT adapter at first.  When i tried to restart Docker, one showed in the network connections, then the error occurred.  After closing the error, it remains, but the next restart gives the same error.  Upon reboot, the adapter is there until docker ties to start.  It then deletes it and gives the error.

Restarting docker again (using the restart link in the control panel BTW) gives the error again and again created the adapter (until the next reboot/restart at least)

![docker_errpr](https://user-images.githubusercontent.com/357843/27133952-fb231ad8-50e1-11e7-89e2-d23794e8e3bd.png)

**Comment by WebMatrixware** [comment]
Similar issue.

```
Checking for common problems
Container Host OS Product Name: Windows 10 Pro
Container Host OS Build Label: 17134.1.amd64fre.rs4_release.180410-1804
Describing Windows Version and Prerequisites
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 686ms
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 134ms
 [+] Is not a build with blocking issues 16ms
 [+] Has 'Containers' feature installed 3.53s
Describing Docker is installed
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  65ms
 [+] Service is running 13ms
 [+] Docker.exe is in path 2.12s
 [+] Docker is registered in the EventLog service 129ms
Describing User has permissions to use Docker daemon
 [+] docker.exe should not return access denied 27ms
Describing Windows container settings are correct
 [+] Do not have DisableVSmbOplock set to 1 28ms
 [+] Do not have zz values set 17ms
 [+] Do not have FDVDenyWriteAccess set to 1 23ms
Describing The right container base images are installed
docker.exe : error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.38/images/json: open //./pipe/docker_engine: The system 
cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This 
error may also indicate that the docker daemon is not running.
At line:106 char:24
+     $imageListOutput = docker.exe images
+                        ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (error during co...is not running.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
 [-] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should be installed 435ms
   ValidationMetadataException: The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
   ParameterBindingValidationException: Cannot validate argument on parameter 'Property'. The argument is null or empty. Provide an argument t
hat is not null or empty, and then try the command again.
   at <ScriptBlock>, <No file>: line 120
Describing Container network is created
 [+] At least one local container network is available 1.08s
 [-] At least one NAT, Transparent, or L2Bridge Network exists 86ms
   Expected {0} to be greater than {0}
   211:       $totalnets | Should BeGreaterThan 0
   at <ScriptBlock>, <No file>: line 211
 [-] NAT Network's vSwitch is internal 39ms
   Expected: {Internal}
   But was:  {}
   216:       $switchType | Should Be "Internal"
   at <ScriptBlock>, <No file>: line 216
 [+] A Windows NAT is configured if a Docker NAT network exists 256ms
 [-] Specified Network Gateway IP for NAT network is assigned to Host vNIC 63ms
   Expected: value to not be empty
   230:       $natGatewayIP | Should Not BeNullOrEmpty
   at <ScriptBlock>, <No file>: line 230
 [-] NAT Network's internal prefix does not overlap with external IP' 36ms
   Expected {0} to be greater than {0}
   257:         $hostips.Count | Should BeGreaterThan 0
   at <ScriptBlock>, <No file>: line 257
Showing output from: docker info

Showing output from: docker version
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:05:28 2018
 OS/Arch:           windows/amd64
 Experimental:      false

Showing output from: docker network ls

Getting Warnings & errors in the Windows event logs from the last 24 hours
Logs saved to C:\WINDOWS\system32\logs_20180827-094418.csv


Getting Docker for Windows daemon logs from the last execution
    Note: More logs may be available at C:\Users\blanning\AppData\Local\Docker. Only showing the latest 100 lines.
```
